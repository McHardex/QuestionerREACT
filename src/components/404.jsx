import React from 'react';

const NOTFOUND = () => (
  <div>
    <h1>Page not found bro</h1>
  </div>
);

export default NOTFOUND;
