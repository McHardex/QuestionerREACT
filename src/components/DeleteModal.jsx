import React from 'react';
import SweetAlert from 'react-bootstrap-sweetalert';
import propTypes from 'prop-types';

const DeleteModal = ({ confirmDelete, closeModal }) => (
  <div className="sweet-alert">
    <SweetAlert
      warning
      showCancel
      confirmBtnText="Yes, delete it!"
      confirmBtnBsStyle="danger"
      cancelBtnBsStyle="default"
      title="Are you sure?"
      onConfirm={confirmDelete}
      onCancel={closeModal}
    >
      You will not be able to recover this Meetup!
    </SweetAlert>
  </div>
);

DeleteModal.propTypes = {
  confirmDelete: propTypes.func.isRequired,
  closeModal: propTypes.func.isRequired,
};

export default DeleteModal;
