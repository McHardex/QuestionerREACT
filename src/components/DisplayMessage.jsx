import React from 'react';
import { CSSTransition } from 'react-transition-group';
import propTypes from 'prop-types';

const DisplayMessage = ({
  error, message, onClick, successClass,
}) => (
  <CSSTransition
    in={error}
    timeout={5000}
    classNames="alert"
    unmountOnExit
  >
    <div className={`error-cont ${successClass}`}>
      <p className="error">{message}</p>
      <span className="exit-error" role="presentation" onClick={onClick}>X</span>
    </div>
  </CSSTransition>
);


DisplayMessage.defaultProps = {
  successClass: '',
  message: null,
  error: false,
};

DisplayMessage.propTypes = {
  error: propTypes.bool,
  message: propTypes.string,
  onClick: propTypes.func.isRequired,
  successClass: propTypes.string,
};

export default DisplayMessage;
