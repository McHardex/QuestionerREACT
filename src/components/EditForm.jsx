import React, { Component } from 'react';
import propTypes from 'prop-types';
import { WithContext as ReactTags } from 'react-tag-input';
import Label from './Label';
import '../assets/stylesheets/editForm.css';


export class EditForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tags: [],
    };
  }

  handleDelete = (i) => {
    const { tags } = this.state;
    this.setState({
      tags: tags.filter((tag, index) => index !== i),
    });
  }

  handleAddition = (tag) => {
    this.setState(state => ({ tags: [...state.tags, tag] }));
  }

  handleDrag = (tag, currPos, newPos) => {
    const { tags: tagsInState } = this.state;
    const tags = [...tagsInState];
    const newTags = tags.slice();

    newTags.splice(currPos, 1);
    newTags.splice(newPos, 0, tag);

    this.setState({ tags: newTags });
  }

  updateMeetup = (e) => {
    e.preventDefault();
    const { tags } = this.state;
    const tagInputs = tags.map(tag => tag.text);
    const data = {
      tags: tagInputs,
    };
    const { target } = e;
    const formData = new FormData(target);
    for (const entry of formData.entries()) {
      const [keys, values] = entry;
      data[keys] = values;
    }

    const { updateMeetup } = this.props;
    updateMeetup(e.target.id, data);
    e.target.reset();
    this.setState({ tags: [] });
  }

  handleTag1 = (e) => {
    e.preventDefault();
    this.setState({ tag1: e.target.value });
    const { tags, tag1 } = this.state;
    tags.push(tag1);
  }

  handleTag2 = (e) => {
    e.preventDefault();
    this.setState({ tag2: e.target.value });
    const { tags, tag2 } = this.state;
    tags.push(tag2);
  }

  handleTag3 = (e) => {
    e.preventDefault();
    this.setState({ tag3: e.target.value });
    const { tags, tag3 } = this.state;
    tags.push(tag3);
  }

  render() {
    const { closeEditModal, singleMeetup } = this.props;
    const { tags } = this.state;

    return (
      <div className="edit-form-cont">
        <form className="edit-form" id={singleMeetup[0] && singleMeetup[0].meetupId} onSubmit={this.updateMeetup}>
          <Label htmlFor="topic">Topic</Label>
          <input
            className="edit-input"
            name="topic"
            type="text"
            defaultValue={singleMeetup[0] && singleMeetup[0].topic}
            required
          />
          <Label htmlFor="date">Date</Label>
          <input
            className="edit-input empty"
            name="happeningOn"
            type="date"
            // defaultValue={convertDate.includes('+') ? removeChar : convertDate}
            required
          />
          <Label htmlFor="location">Location</Label>
          <input
            className="edit-input"
            name="location"
            type="text"
            defaultValue={singleMeetup[0] && singleMeetup[0].location}
            required
          />
          <Label htmlFor="tags">Tags</Label>
          <div className="tags-cont">
            <div className="tags-cont-show">
              <ReactTags
                tags={tags}
                labelField="text"
                placeholder="Add tags"
                handleDelete={this.handleDelete}
                handleAddition={this.handleAddition}
                handleDrag={this.handleDrag}
                maxLength={12}
              />
            </div>
          </div>
          <button type="submit" className="submit-edit">Update Meetup</button>
          <span className="close-edit-tab" onClick={closeEditModal} role="presentation">X</span>
        </form>
      </div>
    );
  }
}

EditForm.defaultProps = {
  singleMeetup: [{}],
};

EditForm.propTypes = {
  closeEditModal: propTypes.func.isRequired,
  updateMeetup: propTypes.func.isRequired,
  singleMeetup: propTypes.arrayOf(propTypes.shape),
};

export default EditForm;
