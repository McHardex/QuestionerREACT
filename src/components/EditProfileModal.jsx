import React from 'react';
import propTypes from 'prop-types';
import Label from './Label.jsx';

const EditProfileModal = ({
  firstname,
  lastname,
  username,
  othername,
  closeModal,
  submitForm,
}) => (
  <div className="profile-details">
    <form className="formCont" onSubmit={submitForm} autoComplete="off">
      <div className="wrap-input">
        <Label className="editProfile-label" htmlFor="Firstname">
          Firstname:
        </Label>
        <input
          type="text"
          className="edit-profile"
          name="firstname"
          defaultValue={firstname}
        />
      </div>
      <div className="wrap-input">
        <Label className="editProfile-label" htmlFor="Lastname">
          Lastname:
        </Label>
        <input
          type="text"
          className="edit-profile"
          name="lastname"
          defaultValue={lastname}
        />
      </div>
      <div className="wrap-input">
        <Label className="editProfile-label" htmlFor="Othername">
          Othername:
        </Label>
        <input
          type="text"
          className="edit-profile"
          name="othername"
          defaultValue={othername}
        />
      </div>
      <div className="wrap-input">
        <Label className="editProfile-label" htmlFor="Username">
          Username:
        </Label>
        <input
          type="text"
          className="edit-profile"
          name="username"
          defaultValue={username}
        />
      </div>
      <button type="submit" className="submit-profile-edit">
        Update Profile
      </button>
    </form>
    <button type="button" className="close-edit-profile" onClick={closeModal}>
      X
    </button>
  </div>
);

EditProfileModal.defaultProps = {
  firstname: null,
  lastname: null,
  username: null,
  othername: null,
};

EditProfileModal.propTypes = {
  firstname: propTypes.string,
  lastname: propTypes.string,
  username: propTypes.string,
  othername: propTypes.string,
  closeModal: propTypes.func.isRequired,
  submitForm: propTypes.func.isRequired,
};
export default EditProfileModal;
