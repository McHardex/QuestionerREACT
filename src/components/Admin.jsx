import React, { Component } from 'react';
import '../assets/stylesheets/admin.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { WithContext as ReactTags } from 'react-tag-input';
import propTypes from 'prop-types';
import Header from './Header';
import Label from './Label';
import { getCurrentUser, getAllMeetups } from '../actions/meetupActions';
import { createMeetup, clearError } from '../actions/adminActions';
import DisplayMessage from './DisplayMessage';
import Loader from './Loader';

class Admin extends Component {
  constructor(props) {
    super(props);

    this.state = {
      searchValue: null,
      length: null,
      meetup: null,
      tags: [],
    };
  }

  handleDelete = (i) => {
    const { tags } = this.state;
    this.setState({
      tags: tags.filter((tag, index) => index !== i),
    });
  }

  handleAddition = (tag) => {
    this.setState(state => ({ tags: [...state.tags, tag] }));
  }

  handleDrag = (tag, currPos, newPos) => {
    const { tags: tagsInState } = this.state;
    const tags = [...tagsInState];
    const newTags = tags.slice();

    newTags.splice(currPos, 1);
    newTags.splice(newPos, 0, tag);

    this.setState({ tags: newTags });
  }

  searchMeetups = (e) => {
    e.preventDefault();
    const searchContent = e.target.value.toLowerCase();
    const { meetups } = this.props;
    const searchResult = meetups.meetups.filter(meetup => (
      meetup.topic.toLowerCase().includes(searchContent)
      || meetup.location.toLowerCase().includes(searchContent)
      || meetup.tags.join(' ').toLowerCase().includes(searchContent)
    ));

    this.setState({
      searchValue: e.target.value,
      length: searchResult.length,
      meetup: searchResult,
    });
  }

  postMeetup = (e) => {
    e.preventDefault();
    const { tags } = this.state;
    const tagInputs = tags.map(tag => tag.text);
    const data = {
      tags: tagInputs,
    };
    const { target } = e;
    const formData = new FormData(target);
    for (const entry of formData.entries()) {
      const [keys, values] = entry;
      data[keys] = values;
    }

    const { createMeetup, getAllMeetups } = this.props;
    createMeetup(data, () => {
      getAllMeetups();
      target.reset();
      this.setState({ tags: [] });
    });
  }

  clearError = () => {
    const { clearError } = this.props;
    clearError();
  }

  componentDidMount = () => {
    const {
      getCurrentUser, getAllMeetups,
    } = this.props;
    getCurrentUser();
    getAllMeetups();
  }

  componentWillReceiveProps = (nextProps) => {
    this.setState({
      meetup: nextProps.meetups.meetups,
    });
  }

  render() {
    const { meetups, admin } = this.props;
    const { message, postMeetupError, isLoading } = admin;
    const { user } = meetups;
    const { isAdmin, username } = user;

    const {
      meetup, length, searchValue, tags,
    } = this.state;

    return (
      <div className="admin-cont">
        <DisplayMessage
          error={postMeetupError}
          message={message}
          onClick={this.clearError}
        />
        <Header role={isAdmin} username={username} />
        <div className="cont">
          <div className="create-meetup-bk">
            <div className="banner-text">
              <h1>Create a new Meetup</h1>
              <p>Enrich lives through open mindset and constructive discussions</p>
            </div>
          </div>
          <div className="cte-meetup-cont">
            <form className="create-meetup" id="create-meetup" onSubmit={this.postMeetup}>
              <Label htmlFor="topic">Topic</Label>
              <input className="form-input" name="topic" type="text" required />
              <Label htmlFor="date">Date</Label>
              <input className="form-input empty" name="happeningOn" type="date" required />
              <Label htmlFor="location">Location</Label>
              <input className="form-input" name="location" type="text" required />
              <Label htmlFor="tags">Tags</Label>
              <div className="tags-cont-show">
                <ReactTags
                  tags={tags}
                  labelField="text"
                  handleDelete={this.handleDelete}
                  handleAddition={this.handleAddition}
                  handleDrag={this.handleDrag}
                  maxLength={12}
                />
              </div>
              <button type="submit" className="submit-form">Create Meetup</button>
            </form>
          </div>
          <hr />
          <div className="search-cont">
            <input
              type="text"
              name="search"
              autoComplete="on"
              placeholder="Search meetup by title, location or tags"
              className="search-meetup"
              onChange={this.searchMeetups}
              tabIndex="-2"
            />
          </div>
          <div id="overlay" />
          <div className="meetups" id="meetups" />
        </div>
        {length === 0
          ? (
            <div className="meetups">
              <p className="not-found">
                Sorry, we could not find any meetups matching
                {' '}
                {`"${searchValue}"`}
              </p>
            </div>
          )
          : (
            <div className="meetups" id="meetups">
              {
                meetup && meetup.map(meetup => (
                  <Link to={`/meetups/${meetup.id}`} className="link" key={meetup.id}>
                    <div
                      className="meetup-cont"
                      role="presentation"
                      key={meetup.id}
                    >
                      <div className="meetup-text">
                        <p className="date">{new Date(meetup.happeningon).toDateString()}</p>
                        <h3 id={meetup.id} className="meetup-topic">{meetup.topic}</h3>
                        <p className="loctn">{meetup.location}</p>
                        <div className="tags-cont">
                          {meetup.tags.map(tag => (
                            <span key={tag}>{tag}</span>
                          ))}
                        </div>
                      </div>
                      <i className="fas fa-trash" title="delete" id={meetup.id} />
                      <i className="far fa-edit" title="edit" id={meetup.id} />
                    </div>
                  </Link>
                ))
              }
            </div>
          )
        }
        {isLoading && <Loader />}
      </div>
    );
  }
}

Admin.propTypes = {
  getCurrentUser: propTypes.func.isRequired,
  getAllMeetups: propTypes.func.isRequired,
  createMeetup: propTypes.func.isRequired,
  clearError: propTypes.func.isRequired,
  meetups: propTypes.objectOf(propTypes.shape).isRequired,
  admin: propTypes.objectOf(propTypes.shape).isRequired,
};

const mapStateToProps = ({ admin, meetups }) => ({ admin, meetups });

export default connect(mapStateToProps, {
  getCurrentUser,
  getAllMeetups,
  createMeetup,
  clearError,
})(Admin);
