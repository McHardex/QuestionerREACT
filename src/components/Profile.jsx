/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { getCurrentUser } from '../actions/meetupActions';
import {
  getQuestionsCount,
  getCommentsCount,
  getUpcomingMeetups,
  updateProfile,
} from '../actions/profileActions';
import Header from './Header';
import EditProfileModal from './EditProfileModal';
import Loader from './Loader';
import '../assets/stylesheets/profile.css';

export class Profile extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showModal: false,
    };
  }

  async componentDidMount() {
    const {
      getCurrentUser,
      getQuestionsCount,
      getCommentsCount,
      getUpcomingMeetups,
    } = this.props;
    await getCurrentUser();
    await getUpcomingMeetups();
    const { meetups } = this.props;
    const { user } = meetups;
    getCommentsCount(user.id);
    getQuestionsCount(user.id);
    console.log(user, '-----');
  }

  toggleModal = () => {
    const { showModal } = this.state;
    this.setState({ showModal: !showModal });
  };

  updateProfile = (e) => {
    e.preventDefault();
    const { target } = e;
    const data = {};
    const formData = new FormData(target);
    for (const entry of formData.entries()) {
      const [keys, values] = entry;
      data[keys] = values;
    }
    const { updateProfile, getCurrentUser } = this.props;
    updateProfile(data, () => {
      this.toggleModal();
      getCurrentUser();
    });
  };

  render() {
    const { meetups, profile, loader } = this.props;
    const { isLoading } = loader;
    console.log(isLoading);
    const { user } = meetups;
    const { upcomingMeetups } = profile;
    const {
      firstname, email, lastname, username, othername,
    } = user;

    const { showModal } = this.state;

    return (
      <div className="admin-cont">
        <Header />
        {isLoading && <Loader />}
        <div className="profile-wrapper">
          {showModal ? (
            <EditProfileModal
              firstname={firstname}
              lastname={lastname}
              username={username}
              othername={othername}
              closeModal={this.toggleModal}
              submitForm={this.updateProfile}
            />
          ) : (
            <div className="profile-details">
              <table>
                <tbody>
                  <tr>
                    <th>Profile</th>
                  </tr>
                  <tr>
                    <td>Firstname:</td>
                    <td className="name">{`${firstname || 'N/A'}`}</td>
                  </tr>
                  <tr>
                    <td>Lastname:</td>
                    <td className="name">{`${lastname || 'N/A'}`}</td>
                  </tr>
                  <tr>
                    <td>Othername:</td>
                    <td className="name">{`${othername || 'N/A'}`}</td>
                  </tr>
                  <tr>
                    <td>Username:</td>
                    <td className="name">{username}</td>
                  </tr>
                  <tr>
                    <td>Email:</td>
                    <td className="email">{email}</td>
                  </tr>
                </tbody>
              </table>
              <i className="fas fa-user-edit" onClick={this.toggleModal} />
            </div>
          )}
          <div className="activities">
            <table>
              <tbody>
                <tr>
                  <th>Activities</th>
                </tr>
                <tr>
                  <td>No of questions posted</td>
                  <td className="questions-posted">{profile.questionCount}</td>
                </tr>
                <tr>
                  <td>No of questions commented on</td>
                  <td className="question-comment">{profile.commentCount}</td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <hr />
        <h1 className="up-header">Upcoming Meetups</h1>
        <div className="meetups" id="meetups">
          {upcomingMeetups
            && upcomingMeetups.map(meetup => (
              <div className="meetup-wrap-landing-page" key={meetup.id}>
                <Link
                  to={`/meetups/${meetup.id}`}
                  className="link"
                  key={meetup.id}
                >
                  <div
                    className="meetup-cont"
                    role="presentation"
                    key={meetup.id}
                  >
                    <div className="meetup-text">
                      <div className="meet-img">
                        <img
                          src="https://res.cloudinary.com/mchardex/image/upload/v1558174712/logo.png"
                          alt="questioner-logo"
                        />
                      </div>
                      <p className="date">
                        {new Date(meetup.happeningon).toDateString()}
                      </p>
                      <h3 id={meetup.id} className="meetup-topic">
                        {meetup.topic}
                      </h3>
                      <p className="loctn">{meetup.location}</p>
                      <div className="tags-cont">
                        {meetup.tags.map(tag => (
                          <span key={tag}>{tag}</span>
                        ))}
                      </div>
                    </div>
                  </div>
                </Link>
              </div>
            ))}
        </div>
      </div>
    );
  }
}

Profile.propTypes = {
  getCurrentUser: propTypes.func.isRequired,
  getQuestionsCount: propTypes.func.isRequired,
  getCommentsCount: propTypes.func.isRequired,
  updateProfile: propTypes.func.isRequired,
  getUpcomingMeetups: propTypes.func.isRequired,
  meetups: propTypes.shape({
    username: propTypes.string,
  }).isRequired,
  profile: propTypes.shape({}).isRequired,
  loader: propTypes.shape({}).isRequired,
};

const mapStateToProps = ({ meetups, profile, loader }) => ({
  meetups,
  profile,
  loader,
});
export default connect(
  mapStateToProps,
  {
    getCurrentUser,
    getQuestionsCount,
    getCommentsCount,
    getUpcomingMeetups,
    updateProfile,
  },
)(Profile);
