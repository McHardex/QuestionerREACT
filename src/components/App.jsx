/* eslint-disable import/extensions */
import React from 'react';
import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../lib/store';
import LandingPage from './LandingPage';
import Signup from './Signup';
import Login from './Login';
import Admin from './Admin';
import Meetup from './Meetup';
import MeetupDetails from './MeetupDetails';
import Profile from './Profile';
import NOTFOUND from './404';

const App = () => (
  <div>
    <Provider store={store}>
      <Router>
        <div>
          <Switch>
            <Route exact path="/" component={LandingPage} />
            <Route exact path="/signup" component={Signup} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/meetups" component={Meetup} />
            <Route exact path="/admin" component={Admin} />
            <Route exact path="/meetups/:id" component={MeetupDetails} />
            <Route exact path="/profile" component={Profile} />

            <Route path="/signup" component={Signup} />
            <Route path="/login" component={Login} />
            <Route path="/meetups" component={Meetup} />
            <Route path="/admin" component={Admin} />
            <Route path="/not-found" component={NOTFOUND} />
            <Redirect to="/not-found" />
          </Switch>
        </div>
      </Router>
    </Provider>
  </div>
);

export default App;
