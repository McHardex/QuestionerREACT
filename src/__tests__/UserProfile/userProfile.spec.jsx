import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Profile } from '../../components/Profile';
import mock from '../../utils/testMock';

const { profileProps } = mock;

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({});

describe('profile component', () => {
  it('snapshot', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Profile {...profileProps} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call clearError', () => {
    const getCurrentUser = jest.fn();
    shallow(<Profile {...profileProps} />);
    expect(getCurrentUser).not.toHaveBeenCalled();
  });
});
