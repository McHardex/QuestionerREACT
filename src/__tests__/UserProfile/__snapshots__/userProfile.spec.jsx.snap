// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile component snapshot 1`] = `ShallowWrapper {}`;
